inlong-sort/sort-connectors/jdbc/src/main/java/org/apache/inlong/sort/jdbc/internal/JdbcBatchingOutputFormat.java

```diff
@@ -0,0 +1,463 @@
+                                            if (metricData.getNumRecordsOut() != null) {
+                                                metricData.getNumRecordsOut().inc(rowSize);
+                                            }
+                                            if (metricData.getNumRecordsOut() != null) {
+                                                metricData.getNumBytesOut()
+                                                        .inc(dataSize);
+                                            }
+                                            resetStateAfterFlush();
+                                        } catch (Exception e) {
+                                            if (metricData.getDirtyRecords() != null) {
+                                                metricData.getDirtyRecords().inc(rowSize);
+                                            }
+                                            if (metricData.getDirtyBytes() != null) {
+                                                metricData.getDirtyBytes().inc(dataSize);
+                                            }
+                                            resetStateAfterFlush();
+                                            flushException = e;
+                                        }
+                                    }
+                                }
+                            },
+                            executionOptions.getBatchIntervalMs(),
+                            executionOptions.getBatchIntervalMs(),
+                            TimeUnit.MILLISECONDS);
+        }
+    }
+
+    private JdbcExec createAndOpenStatementExecutor(
+            StatementExecutorFactory<JdbcExec> statementExecutorFactory) throws IOException {
+        JdbcExec exec = statementExecutorFactory.apply(getRuntimeContext());
+        try {
+            exec.prepareStatements(connectionProvider.getConnection());
+        } catch (SQLException e) {
+            throw new IOException("unable to open JDBC writer", e);
+        }
+        return exec;
+    }
+
+    private void checkFlushException() {
+        if (flushException != null) {
+            throw new RuntimeException("Writing records to JDBC failed.", flushException);
+        }
+    }
+
+    @Override
+    public final synchronized void writeRecord(In record) throws IOException {
+        checkFlushException();
+
+        rowSize++;
+        dataSize = dataSize + record.toString().getBytes(StandardCharsets.UTF_8).length;
+        try {
+            addToBatch(record, jdbcRecordExtractor.apply(record));
+            batchCount++;
+            if (executionOptions.getBatchSize() > 0
+                    && batchCount >= executionOptions.getBatchSize()) {
+                flush();
+                if (metricData.getNumRecordsOut() != null) {
+                    metricData.getNumRecordsOut().inc(rowSize);
+                }
+                if (metricData.getNumRecordsOut() != null) {
```

Comment:
`metricData.getNumRecordsOut() != null` should be `metricData.getNumBytesOut()`

Reply:
OK